client/java-spring-boot-autoconfigure/src/main/java/com/linecorp/centraldogma/client/spring/CentralDogmaSettings.java

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.centraldogma.client.spring;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.annotation.Nullable;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.PositiveOrZero;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.validation.annotation.Validated;
+
+import com.google.common.base.MoreObjects;
+
+/**
+ * Settings for a Central Dogma client.
+ *
+ * <h3>Example with {@code "host"} option</h3>
+ * <pre>{@code
+ * centraldogma:
+ *   hosts:
+ *   - replica1.examples.com:36462
+ *   - replica2.examples.com:36462
+ *   - replica3.examples.com:36462
+ *   accessToken: appToken-cffed349-d573-457f-8f74-4727ad9341ce
+ *   healthCheckIntervalMillis: 15000
+ *   useTls: false
+ * }</pre>
+ *
+ * <h3>Example with {@code "profile"} option</h3>
+ * <pre>{@code
+ * centraldogma:
+ *   profile: beta
+ *   accessToken: appToken-cffed349-d573-457f-8f74-4727ad9341ce
+ *   healthCheckIntervalMillis: 15000
+ *   useTls: false
+ * }</pre>
+ *
+ * <p>Note that {@code "healthCheckIntervalMillis"} and {@code "useTls"} are optional.</p>
+ */
+@ConfigurationProperties(prefix = "centraldogma")
+@Validated
+public class CentralDogmaSettings {
+    /**
+     * The Central Dogma client profile name. If not specified, {@code "hosts"} must be specified.
+     */
+    @Nullable
+    private String profile;
+
+    /**
+     * The list of Central Dogma hosts. e.g. {@code "foo.example.com"} or {@code "foo.example.com:36462"}.
+     * If not specified, {@code "profile"} must be specified.
```

Comment:
IMO, This restriction can be expressed by JSR validator.
If so, spring boot will print human friendly error report.

Reply:
Good to see.
https://spring.io/blog/2018/03/28/property-binding-in-spring-boot-2-0#origin-support

Reply:
Just added a check for `hosts` and `profile` in the auto configuration implementation.